Keyboards/KeyboardsBase/KeyboardKeys.swift

```diff
@@ -254,31 +254,60 @@ class KeyboardKey: UIButton {
   /// Adjusts the width of a key if it's one of the special characters on the iPad keyboard.
   func adjustPadKeyWidth() {
     if usingExpandedKeyboard {
-      scalarSpecialKeysWidth = (disableAccentCharacters && keyboardState != .symbols) ? 2.2 : 1.0
-      scalarReturnKeyWidth = (disableAccentCharacters && keyboardState != .symbols) ? 2.2 : 1.0
+      // Switch case for controller language
+      switch (controllerLanguage){
+      case "Spanish":
+        scalarCapsLockKeyWidth = (disableAccentCharacters && keyboardState != .symbols) ? 2.0 : 1.2
+        scalarSpecialKeysWidth = (disableAccentCharacters && keyboardState != .symbols) ? 2.2 : 1.0
+        scalarReturnKeyWidth = (disableAccentCharacters && keyboardState != .symbols) ? 1.7 : 1.3
+        //scalarReturnKeyWidth = 1.3
+        scalarShiftKeyWidth = 1.8
+      case "German", "Swedish":
+        scalarCapsLockKeyWidth = 1.8
+        scalarReturnKeyWidth = 1.3
+        scalarSpecialKeysWidth = (disableAccentCharacters && keyboardState != .symbols) ? 2.2 : 1.0
+        scalarReturnKeyWidth = (disableAccentCharacters && keyboardState != .symbols) ? 2.2 : 1.0
+      case "French":
+        scalarCapsLockKeyWidth = 1.2
+        scalarReturnKeyWidth = (disableAccentCharacters && keyboardState != .symbols) ? 1.4 : 1.0
+        scalarShiftKeyWidth = 1.8
+      case "Italian":
+        scalarSpecialKeysWidth = (disableAccentCharacters && keyboardState != .symbols) ? 2.2 : 1.0
+        scalarReturnKeyWidth = (disableAccentCharacters && keyboardState != .symbols) ? 1.5 : 1.0
+        scalarCapsLockKeyWidth = 1.3
+        scalarShiftKeyWidth = 1.8
+      default:
+        scalarSpecialKeysWidth = (disableAccentCharacters && keyboardState != .symbols) ? 2.2 : 1.0
+        scalarReturnKeyWidth = (disableAccentCharacters && keyboardState != .symbols) ? 2.2 : 1.0
+      }
+      //scalarSpecialKeysWidth = (disableAccentCharacters && keyboardState != .symbols) ? 2.2 : 1.0
+      //scalarReturnKeyWidth = (disableAccentCharacters && keyboardState != .symbols) ? 2.2 : 1.0
```

Comment:
nit - Would it make sense to already check `(disableAccentCharacters && keyboardState != .symbols)` just once before and then simply use the boolean result in the `switch` statement? The boolean statements for the ternary operators here are simply repeated really. 

OR.. 
(can most definitely be a TODO for later but..) would it be possible to abstract the logic for the values to the specific language somewhere else? So that only _something like_ the below is possible with no `switch` statement needed? 
```
        scalarCapsLockKeyWidth = currentLanguage.scalarCapsLockKeyWidth 
        scalarReturnKeyWidth = currentLanguage.scalarReturnKeyWidth
        scalarSpecialKeysWidth = currentLanguage.scalarSpecialKeysWidth
        scalarReturnKeyWidth = currentLanguage.scalarReturnKeyWidth
```
Scribe-iOS isn't my domain, so let me know if I'm talking crazy :laughing:

Reply:
Either sounds good to me! What are your thoughts on this, @henrikth93? 

And all of Scribe is your domain, @wkyoshida 😊🤗

Reply:
I think Will's idea of abstracting it is good, and that is something I can try and solve. @wkyoshida I am not sure I understand what you mean regarding the ternary operators, but do you mean that just checking the statement once would be less consuming? To tell the truth, my understanding of ternary operators and what they do under the hood is lacking.

Reply:
@wkyoshida @andrewtavis I think the logic for setting the values could be moved to interfacevariables maybe

Reply:
Hey! :wave: 

> I am not sure I understand what you mean regarding the ternary operators, but do you mean that just checking the statement once would be less consuming?

Oh no, not really :smile: I just meant that since the boolean statements for them is the same, could just write it once, store the result in a variable, then consult the variable in each ternary. It's really just more of a visual suggestion to declutter the code a bit to make it more visually clean - that's all.

---

Also, regarding the abstraction - I'm good with leaving it for another PR, so no stress (unless you'd really like to tackle it here already, then by all means :smile:)

Reply:
Happy to move that to a new PR/issue or have it be committed here. @henrikth93, let me know which one you'd prefer and I'll get to the review soon if we're leaving as is 😊

Reply:
Let us move that to a new issue! :)